core/win32/os.c

```diff
@@ -5759,6 +5759,23 @@ mark_page_as_guard(byte *pc)
     ASSERT(res);
 }
 
+/* Removes guard protection from page containing pc */
+bool
+unmark_page_as_guard(byte *pc, uint prot)
+{
+    uint old_prot;
+    int res;
+    byte *start_page = (byte *)ALIGN_BACKWARD(pc, PAGE_SIZE);
+
+    uint flags = memprot_to_osprot(prot & ~MEMPROT_GUARD);
+    res = protect_virtual_memory(start_page, PAGE_SIZE, flags, &old_prot);
+    /* it is likely that another thread accessed the guarded page
```

Comment:
The comment was not on the right line since it refers to the return instruction and not to the assert which was copied from other uses of protect_virtual_memory.